added report (#7)
willtsai/dlh-sp23-team53: report/dl4h-sp23-project-team53.tex

However, intuition and data indicate that there are in fact hidden patterns
within the combinations of disease diagnoses that may be useful for predicting
future outcomes for patients, but yet have not been leveraged in existing
best-in-class healthcare deep learning models. The Chet model is able to learn
how diagnosed diseases develop over the course of each patient's doctor visits
and then utilize this learned disease combination context to predict future
outcomes and diagnoses. 
The most innovative part of the approach is the design to include both
disease combinational information and the dynamic scheme of disease into the model. To include 
disease combinational information, the paper constructed a weighted disease combination
based on the entire longitudinal EHR data globally and also a disease subgraph based on the 
specific visit locally. To include dynamic scheme of diseases, the paper utilized a disease-level
temporal learning with multiple diagnosis roles and corresponding transition functions to extract
historical contexts.

\section{Scope of reproducibility}
In our reproduction study, we will use the same methodology as proposed by the

> authors for data selection, cleaning, and preprocessing as the authors of the
> paper. Specifically, we will join the MIMIC-III \cite{mimic3} and MIMIC-IV

Reviewer: ```suggestion
authors for data selection, cleaning, and preprocessing. Specifically, we will join the MIMIC-III \cite{mimic3} and MIMIC-IV
```

\cite{mimic4} datasets along the same overlapping time ranges that the authors
describe and split training/validation/test sets randomly using the same ratios
they used. We will build the diagnosis graphs and calculate the adjacency
matrices for their corresponding subgraphs using the same methodology described
by the authors. We will train the model and at least one listed baseline model for diagnosis prediction and hearth failure prediction respectively and compare the performance.
\subsection{Addressed claims from the original paper}
% \# TODO:
% Clearly itemize the claims you are testing:
\begin{itemize}
    \item By utilizing disease combinational information and the dynamic scheme of diseases, the Chet model has higher accuracy for diagnosis prediction and hearth failure prediction than RETAIN model(one of the baseline models). 
    \item The designed global disease graph and visit subgraphs can integrate global and local context from disease combinations to inform the deep learning model, so the Chet model has higher accuracy for diagnosis prediction and hearth failure prediction than $Chet_d-$ model where dynamic part of GNN is removed in Chet.

>     \item The proposed three diagnosis roles and corresponding transition functions can extrat historical context and learn the disease development schemes, so the Chet model has higher accuracy for diagnosis prediction and hearth failure prediction than $Chet_t-$ model where transaction functions are removed in Chet.

Reviewer: ```suggestion
    \item The proposed three diagnosis roles and corresponding transition functions can extract historical context and learn the disease development schemes, so the Chet model has higher accuracy for diagnosis prediction and heart failure prediction than $Chet_t-$ model where transaction functions are removed in Chet.
    \item  The proposed three diagnosis roles and corresponding transition functions can extract historical context and learn the disease development schemes, so the Chet model has higher accuracy for diagnosis prediction and heart failure prediction than $Chet_{att}-$ model where attention from the final patient embedding layer is removed in Chet. This is an additional ablation study we proposed.
```

\subsection{Hyperparameters}
\# TODO:
Describe how you set the hyperparameters and what the source was for their value (e.g. paper, code or your guess). 

\subsection{Implementation}
In our preliminary reproduction implementation, we built a \href{https://github.com/willtsai/dlh-sp23-team53/blob/main/notebook.ipynb}{python notebook} to build a complete flow including hyperparameters setting, 
data preprocessing, data loading, model building, model training and evaluation. For now, we kept the same hyperparameters as the original paper and reused the \href{https://github.com/LuChang-CS/Chet}{author's code} in data preprocessing.
Our major code efforts went into model rebuilding and the training/validation flow. In training/validation part, we built our own training and validation method to streamline the training, validation and test process while reusing the existing 

> schedulars and metrics. In model rebuilding part, we tried to follow closely with the model structure and all the equations in the paper step by step.

Reviewer: ```suggestion
schedulers and metrics. In model rebuilding part, we tried to follow closely with the model structure and all the equations in the paper step by step.
```
